tests/common/libvm.sh

```diff
@@ -17,6 +17,31 @@
 # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
 
+# prepares the VM and library for action
+vm_setup() {
+
+  # If there's already an ssh-config, just use that one. The user might have
+  # created it for a self-provisioned machine. Otherwise, let's just assume
+  # we're using vagrant and generate an ssh-config.
+  if [ ! -f ssh-config ]; then
+    vagrant ssh-config > "${topsrcdir}/ssh-config"
+  fi
+
+  local sshopts="-F ${topsrcdir}/ssh-config \
+                 -o ControlMaster=auto \
+                 -o ControlPath=${topsrcdir}/ssh.sock \
+                 -o ControlPersist=yes"
+  export SSH="ssh $sshopts vmcheck"
+  export SCP="scp $sshopts"
+}
+
+vm_rsync() {
+  pushd ${topsrcdir}
+  rsync -az --no-owner --no-group --filter ":- .gitignore" \
+    -e "ssh -F ssh-config" --exclude .git/ . vmcheck:/root/sync
```

Comment:
Hmm...this duplicates the Vagrant syncd folders bits.   It's an unavoidable cost of doing things this way I guess, but can you add a comment to the `Vagrantfile` or so?